### ISSUE TITLE
Users leaving group chat channel after while

### ISSUE THREAD

**Issue by gardar** [bitlbee/bitlbee-facebook#104]
Users seem to leave open groupchat channels after a while.

Re-joining the channel does not seem to bring them back and adding the users seems to be have done by a invite or by interacting with the groupchat on facebook/messenger.

**Comment by dequis** [comment]
Is this related to #95?

**Comment by gardar** [comment]
Might be related but differs somewhat, the contacts in question are in my friend list and cycling the channel does not bring the users back.

**Comment by dequis** [comment]
If you disconnect and reconnect the account, do they come back?

How often does this happen?

How long does it take for them to leave? Seconds? Minutes? Tens of minutes?

**Comment by gardar** [comment]
> If you disconnect and reconnect the account, do they come back?
Disconnect and reconnect the account does not seem to bring them back.

> How often does this happen?
This seems to happen on all my group chats.

> How long does it take for them to leave? Seconds? Minutes? Tens of minutes?
It takes few hours of a idle chat for them to leave.

Is this maybe expected behaviour when idle?

**Comment by dequis** [comment]
>Is this maybe expected behaviour when idle?

Definitely not expected.

How many groupchat members are affected by this? One, a few, all?

>adding the users seems to be have done by a invite or by interacting with the groupchat on facebook/messenger

I don't understand this part, could you rephrase it?

**Comment by gardar** [comment]
All of the members are affected, only me and the root user are left in the channel.

What I meant is that if I send something to the groupchat while only I and root are there it does not get delivered to the groupchat on facebook.
If open up the groupchat on facebook.com/messenger.com/messenger mobile app and send something to it there, it gets delivered to the groupchat and the groupchat users are then back on the bitlbee groupchat channel.
Also when I'm the only user left (along with root) on the groupchat channel, I am able to invite the users back to the channel.

One theory that just came to my mind, are the users possibly leaving the channel when they go offline on facebook?
If that's the case then this behaviour might be correct. However it would be good to be able to send a message to the groupchat channel that would get deliered to those offline users, rather than having to open facebook.com and send a message to the group there.

**Comment by dequis** [comment]
So all of them leave at once? Can you see their quit or part messages in the channel window? Are they quits or parts?

Is the topic cleared too?

What irc client are you using?

Are there any messages in &bitlbee at the time those users leave?

Post output of `account facebook set` in &bitlbee, remove username.

**Comment by liskin** [comment]
I'm having this problem as well with the difference that even though I'm alone there, messages do get sent to the group chat on Facebook. So I just don't care. :-)

**Comment by dequis** [comment]
@gardar poke

**Comment by gardar** [comment]
@dequis sorry for the late reply, been watching this behaviour to get be able to provide better answers to your questions.

It seems like sometimes the users get the messages even though they are not visible / have left the channel, but sometimes they do not get the messages after they have left the channel.
I'm not sure if it has something to do with how long it has been since they left the channel or not.

The topic is not Cleared.

The users don't leave all at the same time, I get the quit, leaving message on the channel.
They are probably leaving the channel when they go offline on facebook/messenger.

The client I'm using is weechat but I have a znc bouncer between the weechat client and the bitlbee server.

09:12:00 @gardar | account facebook set             
09:12:00   @root | auto_connect = `true'            
09:12:00   @root | auto_reconnect = `true'          
09:12:00   @root | away is empty                    
09:12:00   @root | group_chat_open = `all'          
09:12:00   @root | mark_read = `false'              
09:12:00   @root | mark_read_reply = `true'         
09:12:00   @root | nick_format is empty             
09:12:00   @root | nick_source = `handle'           
09:12:00   @root | password = `*****************' (hidden)   
09:12:00   @root | show_unread = `false'            
09:12:00   @root | sync_interval = `30'             
09:12:00   @root | tag = `facebook'                 
09:12:00   @root | username = *REMOVED*



I wonder if it would be possible to keep all the facebook users online all the time? And just voice/devoice them based on their online/offline status.
That would also help when you want to message offline users from &bitlbee
But that might be something up to bitlbee rather than bitlbee-facebook.

**Comment by dequis** [comment]
Maybe it's about setting `set offline_user_quits false` in &bitlbee
